coreneuron/utils/reports/binary_report_handler.cpp

```diff
@@ -0,0 +1,129 @@
+#include "binary_report_handler.h"
+#include "coreneuron/utils/reports/nrnsection_mapping.h"
+#ifdef ENABLE_REPORTING
+#include "reportinglib/Records.h"
+#endif  // ENABLE_REPORTING
+
+namespace coreneuron {
+
+
+void BinaryReportHandler::create_report(double dt, double tstop, double delay) {
+#ifdef ENABLE_REPORTING
+    records_set_atomic_step(dt);
+#endif  // ENABLE_REPORTING
+    ReportHandler::create_report(dt, tstop, delay);
+}
+
+#ifdef ENABLE_REPORTING
+void BinaryReportHandler::register_soma_report(NrnThread& nt, ReportConfiguration& config,
+                                                VarsToReport& vars_to_report) {
+    int sizemapping = 1;
+    int extramapping = 5;
+    // first column i.e. section numbers
+    int mapping[] = {0};
+    // first row, from 2nd value (skip gid)
+    int extra[] = {1, 0, 0, 0, 0};
+
+    VarsToReport::iterator it;
+    for (it = vars_to_report.begin(); it != vars_to_report.end(); ++it) {
```

Comment:
Yes! this was how was the old code, i just restructured the code to introduce the new reports.
But i guess it makes sense to change this things since are really quickly!